x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -2003,6 +2006,31 @@ private void reloadRemoteClusterCredentials(Settings settingsWithKeystore) {
         future.actionGet();
     }
 
+    public Map<String, String> getAuthContextForSlowLog() {
+        if (this.securityContext.get() != null && this.securityContext.get().getAuthentication() != null) {
+            Authentication authentication = this.securityContext.get().getAuthentication();
+            Subject authenticatingSubject = authentication.getAuthenticatingSubject();
+            Subject effetctiveSubject = authentication.getEffectiveSubject();
+            Map<String, String> authContext = new HashMap<>();
+            if (authenticatingSubject.getUser() != null) {
+                authContext.put("user.name", authenticatingSubject.getUser().principal());
+                authContext.put("user.realm", authenticatingSubject.getRealm().getName());
```

Comment:
We should also add conditionally add `user.full_name`= `authenticatingSubject.getUser().fullName()` if not null. Specifically, SAML is the only way to authenticate in serverless and via the SAML realm the principal is just an internal ID. SAML is also the way most Cloud SSO works. 

For example, here is my info from Cloud (viewable via `GET _security/_authenticate`):
```
"principal": "1583834402",
"full_name": "jake.landis@elastic.co"
```

Reply:
Ahh yes, great catch!